lib/new_relic/agent/tracer.rb

```diff
@@ -409,15 +410,18 @@ def clear_state
 
         alias_method :tl_clear, :clear_state
 
-        def thread_block_with_current_transaction(*args, &block)
+        def thread_block_with_current_transaction(*args, segment_name:, parent:, &block)
```

Comment:
How come parent is passed in as a req kwarg now? Was there an issue with the way parents are decided when we are just simply copying the current txn over?

Reply:
Yes, none of the parents were getting assigned anymore for threads or `Concurrent/Task` segments. They all showed up as fragmented spans, so they'd be within the trace, but not nested or listed in the correct order. Passing the parent as an arg fixed the inheritance issue. Does this seem like it might have ramifications I'm not anticipating?

We may have other thread-instrumentation-related problems on our hands. When @hannahramadan  and I were testing the Concurrent Ruby instrumentation with thread tracing turned on in a Rails app, the Rails app couldn't boot. She's looking into that now. I'm not sure if the `parent` addition is related.

Reply:
Confirming this appears to be the addition that stalls Rails apps with `thread.tracing` enabled.

Reply:
@hannahramadan found the fix! The issue was moving `capture_segment_error` to outside the `if current_txn` condition.